airflow/providers/amazon/aws/operators/ec2.py

```diff
@@ -183,22 +183,36 @@ def execute(self, context: Context):
             MaxCount=self.max_count,
             **self.config,
         )["Instances"]
-        instance_ids = []
-        for instance in instances:
-            instance_ids.append(instance["InstanceId"])
-            self.log.info("Created EC2 instance %s", instance["InstanceId"])
+
+        instance_ids = self._on_kill_instance_ids = [instance["InstanceId"] for instance in instances]
```

Comment:
This type of clean up in case of an error is very similar to the purpose of [context/with statements in python](https://realpython.com/python-with-statement/#the-with-statement-approach)

`on_kill` is called outside the normal execution order as part of the signal handling for `SIGTERM` - hence the extra variable only for that purpose.

Yes, one could call it `self.instance_ids` and I considered it - this depends on the lifecycle of the TaskInstance object, call order of the methods. So the list of instances to clean up should be cleared when XCom has communicated the result. I named the variable very purpose specific.

Would you think it would be necessary to delete the `_on_kill_instance_ids`, e.g. as part of the `post_execute` method?

Reply:
Hmm, I honestly didn't consider that perspective, but I think your approach is correct - better to be cautious and have a potentially redundant variable rather than introducing a race condition of some sort. Thanks for explaining your thoughts, I appreciate it!